packages/jsapi-components/src/useGetItemIndexByValue.ts

```diff
@@ -0,0 +1,39 @@
+import { useCallback } from 'react';
+import { dh } from '@deephaven/jsapi-types';
+import { useTableUtils } from './useTableUtils';
+
+/**
+ * Returns a function that gets the index of the first row containing a column
+ * value.
+ * @param columnName The name of the column to search
+ * @param value The value to search for
+ * @param table The table to search in
+ * @returns A function that returns the index of the first row containing the
+ * matching value, or `null` if no match is found
+ */
+export function useGetItemIndexByValue<TValue>({
+  columnName,
+  value,
+  table,
+}: {
+  columnName: string;
+  table: dh.Table | null;
+  value: TValue | null | undefined;
+}): () => Promise<number | null> {
+  const tableUtils = useTableUtils();
+
+  return useCallback(async () => {
+    if (table == null || value == null) {
+      return null;
+    }
+
+    const column = table.findColumn(columnName);
+    const columnValueType = tableUtils.getValueType(column.type);
+
+    const index = await table.seekRow(0, column, columnValueType, value);
+
+    return index === -1 ? null : index;
```

Comment:
If I'm understanding correctly, the `-1` should be returned for non-matches once deephaven/deephaven-core/pull/5238 is merged. For now, the non-match scenario will return an actual row index. In cases where a table ticks causing a previous selection to drop from the table, this will cause scrolling to a seemingly arbitrary position, but this should be enough of an edge case to not be too worried about it since it is temporary